gwnr/waveform/hybridize.py

```diff
@@ -30,44 +30,70 @@
 import scipy.optimize 
 from scipy.integrate import cumulative_trapezoid
 
-def find_value_location_in_series(frq_timeseries, frq_desired,
-                                  tol):  # More outside the main fn
 
-    if frq_desired < np.min(frq_timeseries):
-        raise Exception(
-            'Desired frequency out of bounds, lower than min frequency')
+def find_last_value_location_in_series(frq_timeseries, frq_desired, tol=None):
+    reversed_freq_timeseries = frq_timeseries[::-1]
+    final_idx = len(reversed_freq_timeseries) - 1
+    for idx, f_value in enumerate(reversed_freq_timeseries):
+        if idx != len(reversed_freq_timeseries) - 1:
+            if reversed_freq_timeseries[idx] >= frq_desired and reversed_freq_timeseries[
+                idx+1] <= frq_desired:
+                fr1 = reversed_freq_timeseries[idx]
+                fr2 = reversed_freq_timeseries[idx + 1]
+                
+                if abs(frq_desired - fr1) <= abs(frq_desired - fr2):
+                    final_idx = idx
+                else:
+                    final_idx = idx + 1
+                break
+    return len(frq_timeseries) - 1 - final_idx
 
-    if frq_desired > np.max(frq_timeseries):
-        raise Exception(
-            'Desired frequency out of bounds, higher than max frequency')
+find_value_location_in_series = find_last_value_location_in_series
```

Comment:
delete this line, and use the new name `find_last_value_location_in_series` below. It was only a temporary hack to get the rest of code in this file to work without having to rename the function calls.